src/game-engine/GameEngineA.ts

```diff
@@ -147,10 +147,16 @@ export default class GameEngineA {
   choosePlay(aPlay: Play) {
     if (!(this.state instanceof State.ReadyToChooseAPlay)) { return this.state };
 
-    this.validateBalances()
-
     const { balances, turnNum, stake, channel, adjudicator } = this.state;
 
+    if (this.state.stake > this.state.balances[0]) {
+      return this.transitionTo(
+        new State.InsufficientFundsA({
+          channel, balances, adjudicator,
+        })
+      )
+    }
```

Comment:
You're running this check at the point that A makes their choice. This made me wonder whether this is a good user experience, or whether we could let the user know earlier. It seems like we know in the resting state whether or not another round is possible or not. Do you think it would be preferable to transition to `InsufficentFundsA` instead of `ReadyToChooseAPlay`?

Also, should we check that B also has sufficient funds here?

Reply:
I'm not sure about the UX, and took the lazy strategy of waiting for the last moment. It probably makes more sense to transition to `InsufficentFundsA` over `ReadyToChooseAPlay`, though, and notify the user as early as possible.

>Also, should we check that B also has sufficient funds here?
I'm open to this, but if we do, we need to add another view, and another state: `PlayerA.InsufficientFundsB`, or something like that.